lein midje fails with 3.1.2

When I run lein midje I have this error java.lang.ClassNotFoundException: leiningen.core.main (see error details below)
My configuration 
Leiningen 2.3.2 / lein-midje 3.1.2 / midje 1.5.1 
The same configuration works fine with lein-midje 3.1.1

Exception in thread "main" java.lang.ClassNotFoundException: leiningen.core.main, compiling:(/tmp/form-init2091574418830435753.clj:1:2423)
    at clojure.lang.Compiler.analyzeSeq(Compiler.java:6567)
    at clojure.lang.Compiler.analyze(Compiler.java:6361)
    at clojure.lang.Compiler.analyze(Compiler.java:6322)
    at clojure.lang.Compiler$BodyExpr$Parser.parse(Compiler.java:5708)
    at clojure.lang.Compiler.analyzeSeq(Compiler.java:6560)
    at clojure.lang.Compiler.analyze(Compiler.java:6361)
    at clojure.lang.Compiler.analyze(Compiler.java:6322)
    at clojure.lang.Compiler$IfExpr$Parser.parse(Compiler.java:2677)
    at clojure.lang.Compiler.analyzeSeq(Compiler.java:6560)
    at clojure.lang.Compiler.analyze(Compiler.java:6361)
    at clojure.lang.Compiler.analyzeSeq(Compiler.java:6548)
    at clojure.lang.Compiler.analyze(Compiler.java:6361)
    at clojure.lang.Compiler.analyze(Compiler.java:6322)
    at clojure.lang.Compiler$BodyExpr$Parser.parse(Compiler.java:5708)
    at clojure.lang.Compiler$LetExpr$Parser.parse(Compiler.java:6009)
    at clojure.lang.Compiler.analyzeSeq(Compiler.java:6560)
    at clojure.lang.Compiler.analyze(Compiler.java:6361)
    at clojure.lang.Compiler.analyze(Compiler.java:6322)
    at clojure.lang.Compiler$BodyExpr$Parser.parse(Compiler.java:5708)
    at clojure.lang.Compiler$FnMethod.parse(Compiler.java:5139)
    at clojure.lang.Compiler$FnExpr.parse(Compiler.java:3751)
    at clojure.lang.Compiler.analyzeSeq(Compiler.java:6558)
    at clojure.lang.Compiler.analyze(Compiler.java:6361)
    at clojure.lang.Compiler.eval(Compiler.java:6616)
    at clojure.lang.Compiler.eval(Compiler.java:6609)
    at clojure.lang.Compiler.load(Compiler.java:7064)
    at clojure.lang.Compiler.loadFile(Compiler.java:7020)
    at clojure.main$load_script.invoke(main.clj:294)
    at clojure.main$init_opt.invoke(main.clj:299)
    at clojure.main$initialize.invoke(main.clj:327)
    at clojure.main$null_opt.invoke(main.clj:362)
    at clojure.main$main.doInvoke(main.clj:440)
    at clojure.lang.RestFn.invoke(RestFn.java:421)
    at clojure.lang.Var.invoke(Var.java:419)
    at clojure.lang.AFn.applyToHelper(AFn.java:163)
    at clojure.lang.Var.applyTo(Var.java:532)
    at clojure.main.main(main.java:37)
Caused by: java.lang.ClassNotFoundException: leiningen.core.main
    at java.net.URLClassLoader$1.run(URLClassLoader.java:202)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.net.URLClassLoader.findClass(URLClassLoader.java:190)
    at clojure.lang.DynamicClassLoader.findClass(DynamicClassLoader.java:61)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:306)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:247)
    at java.lang.Class.forName0(Native Method)
    at java.lang.Class.forName(Class.java:247)
    at clojure.lang.RT.classForName(RT.java:2070)
    at clojure.lang.Compiler$HostExpr.maybeClass(Compiler.java:969)
    at clojure.lang.Compiler$HostExpr.access$400(Compiler.java:747)
    at clojure.lang.Compiler.macroexpand1(Compiler.java:6499)
    at clojure.lang.Compiler.analyzeSeq(Compiler.java:6546)
    ... 36 more
Subprocess failed

--- a/src/leiningen/midje.clj
+++ b/src/leiningen/midje.clj
@@ -11,13 +11,17 @@
           (keyword %))
        filters))
 
-(defn make-load-facts-form [namespace-strings filters]
+(defn make-load-facts-form [project namespace-strings filters]
   (let [true-namespaces (map (fn [nss] `(quote ~(symbol nss)))
-                             namespace-strings)]
+                             namespace-strings)
+        exit-fn (if (or (:eval-in-leiningen project) 
+                        (= (:eval-in project) :leiningen))
+                  `main/exit
+                  `System/exit)]
     `(let [failures# (:failures (midje.repl/load-facts ~@true-namespaces
                                                        ~@(repl-style-filters filters)))]
        (when-not (zero? failures#) 
-         (main/exit 255)))))
+         (~exit-fn (min 255 failures#))))))
 
 (defn make-autotest-form [dirs filters]
   ;; Note: filters with an empty arglist means "use the default".
@@ -159,6 +163,7 @@
         exec-form (if (:autotest? control-map)
                     (make-autotest-form (:autotest-args control-map)
                                         (:filter-args control-map))
-                    (make-load-facts-form (:true-args control-map)
+                    (make-load-facts-form project
+                                          (:true-args control-map)
                                           (:filter-args control-map)))]
     (eval-in-project project exec-form init-form)))

--- a/test/lein_midje/t_lein_midje.clj
+++ b/test/lein_midje/t_lein_midje.clj
@@ -76,5 +76,5 @@
       result => #"set-config-files!.*\[\"config.1\"\s+\"config.2\"\]")))
 
 (fact :metadata "filters passed in"
-  (pr-str (make-load-facts-form [] ["integration" "-slow"]))
+  (pr-str (make-load-facts-form {} [] ["integration" "-slow"]))
   => #"load-facts :integration \(clojure.core/complement :slow\)")

